test/src/crypto.rs

```diff
@@ -13,6 +13,29 @@ use openssl::{
     nid::Nid,
     pkey::{PKey, Public},
 };
+use p384::ecdsa::SigningKey;
+use rfc6979::HmacDrbg;
+use sha2::Sha384;
+
+// Derives a key using a DRBG. Returns (priv, pub_x, pub_y)
+pub fn derive_ecdsa_keypair(seed: &[u8]) -> ([u8; 48], [u8; 48], [u8; 48]) {
```

Comment:
There's a lot of overlap with hmac384_drbg_keygen() in this file, and derive_ecdsa_key() below.